src/base.ts

```diff
@@ -0,0 +1,271 @@
+        const nextPage = getPageId(pageNumber + 1);
+        canvasContainer.scrollTo({ top: (document.getElementById(nextPage)?.parentElement as HTMLElement).offsetTop });
+        pageNumberDiv.textContent = `${pageNumber + 1}`;
+      }
+    };
+    prevButton.onclick = () => {
+      const pageNumber = parseInt(pageNumberDiv.textContent || '1');
+      if (pageNumber > 1) {
+        const prevPage = getPageId(pageNumber - 1);
+        canvasContainer.scrollTo({ top: (document.getElementById(prevPage)?.parentElement as HTMLElement).offsetTop });
+        pageNumberDiv.textContent = `${pageNumber - 1}`;
+      }
+    };
+    nextButton.innerHTML = chevronRight;
+    prevButton.innerHTML = chevronLeft;
+
+    zoomButton.innerHTML = plusSvg;
+    containerDiv.appendChild(controlsDiv);
+    controlsDiv.appendChild(prevButton);
+    controlsDiv.appendChild(fullPageNumberDiv);
+    controlsDiv.appendChild(nextButton);
+    controlsDiv.appendChild(zoomButton);
+    containerDiv.appendChild(canvasContainer);
+    const pages = [...Array(pdfDocument.numPages).keys()];
+
+    // load each page synchronously so that they appear in order
+    while (pages.length) {
+      const page = pages.shift();
+      if (page !== undefined) {
+        // render the PDF page
+        const pageContainer = document.createElement('div');
+        pageContainer.className = 'pageContainer';
+        pageContainer.style.width = defaultWidth;
+        canvasContainer.appendChild(pageContainer);
+        const pdfPage = await pdfDocument.getPage(page + 1);
+        const viewport = pdfPage.getViewport({ scale: pdfScale });
+        const canvas = document.createElement('canvas');
+        canvas.className ='pdfViewerCanvas';
+        canvas.id = getPageId(page + 1);
+        pageContainer.appendChild(canvas);
+        vw.viewerState[documentId]?.canvasElements.push(canvas);
+        canvas.width = viewport.width;
+        canvas.height = viewport.height;
+        canvas.style.width = '100%';
+        const ctx = canvas.getContext('2d');
+        await pdfPage.render({
+          canvasContext: ctx || {},
+          viewport,
+        });
+
+        // initial setup for page breakpoints
+        const breakPoint =
+          (vw.viewerState[documentId]?.pageBreaks[(vw.viewerState[documentId]?.pageBreaks.length || 0) - 1] || - (canvas.offsetHeight / 2 + 16)) + canvas.offsetHeight + 32;
+        vw.viewerState[documentId]?.pageBreaks.push(breakPoint);
+
+        // render text layer for selection purposes
+        const textLayerDiv = document.createElement('div');
+        textLayerDiv.className = 'textLayer';
+        const textContent = await pdfPage.getTextContent();
+        pageContainer.appendChild(textLayerDiv);
+        scalePDF(
+          textLayerDiv,
+          textContent,
+          pdfPage,
+          canvas,
+          viewport,
+          documentId
+        );
+
+        // handle zoom in/out
+        zoomButton.addEventListener('click', () => {
```

Comment:
Since this is not React, these event handlers get cleaned up when the component is removed from the DOM, correct?

Reply:
yep, event listeners are removed with the component. The only ones that don't get removed are on things like the `window` or `document` which aren't reloaded when you change pages etc.